### PR TITLE
feat: Add launch_template_tags variable for additional launch template tags

### PR DESCRIPTION
**Pull request by kahirokunn**
## Description
add "launch_template_ignore_changes" option

## Motivation and Context
If I add a tag to `var.tags`, it will be added to all of them, but there was a situation where I wanted to add an arbitrary tag only to the specifications_tags of launch_template_tag, and I want to make it supported.

### CODE REVIEW

File: modules/eks-managed-node-group/variables.tf [hunk 1]
```diff
@@ -408,6 +408,12 @@ variable "security_group_tags" {
   default     = {}
 }
 
+variable "launch_template_tag_specifications_tags" {
```
Comment (bryantbiggs) [inline modules/eks-managed-node-group/variables.tf:411]:
```suggestion
variable "launch_template_tags" {
```

Comment (bryantbiggs) [reply modules/eks-managed-node-group/variables.tf:411]: you'll need to update in both places and fix the static checks (i.e. - `pre-commit run -a`)

**Comment by bryantbiggs** [comment]
I almost forgot - we'll need to update self-managed node groups module as well, and then propagate those changes  up to the root `node_groups.tf` as well - and the title should be updated to reflect - "feat: Add `launch_template_tags` variable for additional launch template tags"

**Comment by kahirokunn** [comment]
@bryantbiggs
I think I did everything you told me to do.
But `pre-commit run -a` command does not complete no matter how long I wait 😢

**Comment by bryantbiggs** [comment]
do you have pre-commit installed? in this repo it can take quite some time - you can get around this with (executed  from project root):
- `terraform fmt -recursive .`
- `pre-commit run terraform_docs -a`

**Comment by kahirokunn** [comment]
@bryantbiggs 
It was just like you told me.
I think I've accomplished all the work I needed to!
Thx!

File: node_groups.tf [hunk 2]
```diff
@@ -296,6 +296,7 @@ module "eks_managed_node_group" {
   enable_monitoring                  = try(each.value.enable_monitoring, var.eks_managed_node_group_defaults.enable_monitoring, true)
   network_interfaces                 = try(each.value.network_interfaces, var.eks_managed_node_group_defaults.network_interfaces, [])
   placement                          = try(each.value.placement, var.eks_managed_node_group_defaults.placement, null)
+  launch_template_tags               = try(each.value.launch_template_tags, var.eks_managed_node_group_defaults.launch_template_tags, null)
```
Comment (bryantbiggs) [inline node_groups.tf:299]: woops, one more - the very last default should be `{}` like the other tags are set here, and we can move this up to line 276

File: node_groups.tf [hunk 3]
```diff
@@ -420,6 +421,7 @@ module "self_managed_node_group" {
   enable_monitoring                  = try(each.value.enable_monitoring, var.self_managed_node_group_defaults.enable_monitoring, true)
   network_interfaces                 = try(each.value.network_interfaces, var.self_managed_node_group_defaults.network_interfaces, [])
   placement                          = try(each.value.placement, var.self_managed_node_group_defaults.placement, null)
+  launch_template_tags               = try(each.value.launch_template_tags, var.self_managed_node_group_defaults.launch_template_tags, null)
```
Comment (bryantbiggs) [inline node_groups.tf:424]: likewise here, default of `{}` and move up to line 394

File: node_groups.tf (hunk 2 above)
Comment (kahirokunn) [reply node_groups.tf:299]: done

File: node_groups.tf (hunk 3 above)
Comment (kahirokunn) [reply node_groups.tf:424]: done

**Comment by kahirokunn** [comment]
@bryantbiggs how about now?

File: node_groups.tf [hunk 4]
```diff
@@ -273,6 +273,7 @@ module "eks_managed_node_group" {
   launch_template_use_name_prefix = try(each.value.launch_template_use_name_prefix, var.eks_managed_node_group_defaults.launch_template_use_name_prefix, true)
   launch_template_version         = try(each.value.launch_template_version, var.eks_managed_node_group_defaults.launch_template_version, null)
   launch_template_description     = try(each.value.launch_template_description, var.eks_managed_node_group_defaults.launch_template_description, "Custom launch template for ${try(each.value.name, each.key)} EKS managed node group")
+  launch_template_tags            = try(each.value.launch_template_tags, var.eks_managed_node_group_defaults.launch_template_tags, null)
```
Comment (bryantbiggs) [inline node_groups.tf:276]: Change the defaults please (on both)
```suggestion
  launch_template_tags            = try(each.value.launch_template_tags, var.eks_managed_node_group_defaults.launch_template_tags, {})
```

File: node_groups.tf [hunk 5]
```diff
@@ -390,6 +391,7 @@ module "self_managed_node_group" {
   launch_template_use_name_prefix = try(each.value.launch_template_use_name_prefix, var.self_managed_node_group_defaults.launch_template_use_name_prefix, true)
   launch_template_version         = try(each.value.launch_template_version, var.self_managed_node_group_defaults.launch_template_version, null)
   launch_template_description     = try(each.value.launch_template_description, var.self_managed_node_group_defaults.launch_template_description, "Custom launch template for ${try(each.value.name, each.key)} self managed node group")
+  launch_template_tags            = try(each.value.launch_template_tags, var.self_managed_node_group_defaults.launch_template_tags, null)
```
Comment (bryantbiggs) [inline node_groups.tf:394]:
```suggestion
  launch_template_tags            = try(each.value.launch_template_tags, var.self_managed_node_group_defaults.launch_template_tags, {})
```

**Review by bryantbiggs — CHANGES_REQUESTED** [review]

**Comment by kahirokunn** [comment]
The correction was omitted.
Sorry about that.

**Review by bryantbiggs — APPROVED** [review]
thanks for your efforts @kahirokunn 🎉 

good to go 👍🏽 @antonbabenko

**Comment by kahirokunn** [comment]
Thanks for all the advice!
I've learned so much!

**Comment by antonbabenko** [comment]
This PR is included in version 18.3.0 :tada:

**Pull request merged**